_posts/2017-10-04-Good-Bad-And-Ugly.md

```diff
@@ -0,0 +1,62 @@
+
+![Studio](/assets/images/studio.png)
+
+And the use of graphical based elements while creating the application does not stop there. Many of the pages for you application are semi-done for you. Out of the box you get list views, which allow you to view database data in table-like format. You want to view a specific entry in the table? In ServiceNow it's called a record, and you can view it using Form View. You don't like the layout of the form? There is even a form designer allowing you to drag and drop element to customize that!
+
+So far so good, you don't have to play around with such a boring task as creating a table view of your data. But your application does not do anything useful yet. You want to have a flow in the app, you want something to trigger actions. You look up how to do that, since you still have not seen any place for writing any code, and you find out that you should do that using workflows. How to create a workflow? There is a workflow editor for that - a drag and drop application that enables you to create and customize workflows.
+
+I hope at this point you can already see my point. An enormous part of the application is being created using ServiceNow specific GUI applications. You may think that this is a great thing! You don't need to learn another API! You do not need to go through setting up annoying things like database connection, user authorization and authentication, basic table and form views. Some of you may also think that this way of doing stuff is completely absurd! How can I work without my beloved IDE, frameworks and libraries? How can I debug and test my code? You have to know that in ServiceNow things are done in a **different** way.
+### Implementing Code
+Reading this post you may think that you do not need to know how to write code to be able to write an application for ServiceNow. That is wrong. There are places that require coding: business rules, client scripts, script includes just to name few. And if you want to go a step further and write a custom looking application using Service Portal, you need to write JavaScript, HTML and CSS for widgets.
+
+Here I encountered my first actual problem with ServiceNow platform. I was very sceptical about having to build the application using provided interfaces from the beginning, but since it was working well, I am able to justify it for the comfort of having all the nice functionality out of the box. But I can find no particular reason to make developers write code inside textareas on the ServiceNow page. I am aware that this way you could write code using for instance a chromebook, but that is a step back! Why would you cut your grass using a scythe, when there are lawnmowers created for that?
```

Comment:
"write code using for instance" -> "write code user, for instance,"

Reply:
I hope you mean "write code using, for instance,"

Reply:
yes, correct